babel/messages/_compat.py

```diff
@@ -0,0 +1,31 @@
+import sys
+from functools import partial
+
+
+def find_entrypoints(group_name: str):
+    """
+    Find entrypoints of a given group using either `importlib.metadata` or the
+    older `pkg_resources` mechanism.
+
+    Yields tuples of the entrypoint name and a callable function that will
+    load the actual entrypoint.
+    """
+    if sys.version_info >= (3, 10):
+        # "Changed in version 3.10: importlib.metadata is no longer provisional."
+        try:
+            from importlib.metadata import entry_points
+        except ImportError:
```

Comment:
Is the try/except block needed if we have an explicit version check? `importlib.metadata` should always be available on 3.10+

Reply:
> Is the try/except block needed if we have an explicit version check? `importlib.metadata` should always be available on 3.10+

https://github.com/python-babel/babel/pull/1102#discussion_r1684164348

Reply:
As @podgorniy94 mentioned, I think it's a good thing to double-check, for supporting some more esoteric environments.

Reply:
> As @podgorniy94 mentioned, I think it's a good thing to double-check, for supporting some more esoteric environments.

Actually, there's no need to check the Python version. We simply try `pkg_resources`, and if the library isn't available, then we attempt `importlib.metadata`. To avoid strict version checking for, as you said, esoteric environments where there might be different version labeling :)

Reply:
Yes and no – using `pkg_resources` where it's deprecated will raise a DeprecationWarning, which in some (in e.g. my work-work) environments deprecation warnings are hard errors in tests.